### PR TITLE
add new 'Compare to' section under ESC in docs

### PR DESCRIPTION
**Pull request by shughes26**
### Proposed changes

Added new `Compare to` directory under ESC in docs, and page for ESC vs Vault
https://github.com/pulumi/marketing/issues/1060

### CODE REVIEW

File: content/docs/esc/vs/_index.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+Pulumi is a cloud-native infrastructure as code project. It lets you provision and manage resources across many clouds---AWS, Azure, Google Cloud, Kubernetes, OpenStack---using your favorite language. It works great for a wide range of
+cloud infrastructures and applications, including containers, virtual machines, databases, cloud services, and serverless.
+
+Because of this broad array of supported scenarios, there are many tools that overlap with Pulumi's capabilities. Many
+of these are complementary and can be used together, whereas some are "either or" decisions.
+
+Here are several useful comparisons that will help you understand Pulumi's place in the cloud tooling ecosystem:
+
+* [Hashicorp Terraform](/docs/concepts/vs/vault/)
```
Comment (aaronkao) [inline content/docs/esc/vs/_index.md:22]:
```suggestion
* [HashiCorp Vault](/docs/esc/vs/vault/)
```

**Review by aaronkao — CHANGES_REQUESTED** [review]
Pushing through some changes here

**Comment by arunkumar611** [comment]
We should change the get started here at the bottom of the page. It is talking about IaC:

image

File: layouts/shortcodes/get-started-esc.html [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+                </a>
+            </div>
+            <div class="pb-4 md:pr-4 md:w-1/2">
+                <a class="tile p-8" href="/docs/pulumi-cloud/esc/providers/gcp-secrets/">
+                    <img class="h-10 mx-auto" src="/logos/tech/gcp.svg" alt="Google Cloud" />
+                </a>
+            </div>
+            <div class="pb-4 md:w-1/2">
+                <a class="tile p-8" href="/docs/pulumi-cloud/esc/providers/vault-secrets/">
+                    <img class="h-10 mx-auto" src="/logos/tech/terraform.svg" alt="Terraform" />
```
Comment (aaronkao) [inline layouts/shortcodes/get-started-esc.html:28]: This should be Vault and not Terraform

**Review by aaronkao — APPROVED** [review]

**Pull request merged**